move-to-sdk/app/io/sphere/sdk/facets/FacetOptionMapper.java

```diff
@@ -0,0 +1,17 @@
+package io.sphere.sdk.facets;
+
+import java.util.List;
+
+/**
+ * Mapper class that allows to transform the facet options, as they are extracted from the search result, into another list.
+ * For example, to transform a list of category IDs into a hierarchical structure of categories with localized names.
+ */
+public interface FacetOptionMapper {
```

Comment:
Actually initially it was a UnitaryOperator when it was only for the CategoryHierarchy. Then I realized it became super big and you had to put everything in a class anyway. But then I didn't have a direct way to let developers where to find these UnitaryOperators, so I had to allow one method to accept a `CategoryHierarchyFacetOptionMapper`... and then I thought these mappers could be used heavily for multiple things... that's when the `FlexibleSelectFacet` was born, which accepts a mapper to manipulate the options, and ideally I would like to offer most of the typical manipulations via this Mapper.